### PR TITLE
add blogpost: dhcp subnet data

### PR DESCRIPTION
**Pull request by timogoebel**
This adds some info about a technique which I've shown at the Foreman Birthday Event in Nuremberg.

@GregSutcliffe: You wanted this. What do you think? :-)

cc: @martinpfeifer

### CODE REVIEW

File: _posts/2016-07-22-populating-your-dhcp-server-with-subnet-data-from-foreman.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+tags:
+- foreman
+- puppet
+- dhcp
+- subnets
+---
+
+ISC's DHCP-server is widely used on Linux systems to offer dynamic IP addresses to computers on a network. DHCP plays an important part in the PXE provisioning workflow.
+When you use Foreman and DHCP to manage your network's IP addresses, you usually have to define subnets twice: Once in Foreman and once in `/etc/dhcp/dhcpd.conf` on your server.
+This is tedious and error prone. There is a way to import subnets from a DHCP server into Foreman, however you might want to to it the other way round.
```
Comment (GregSutcliffe) [inline _posts/2016-07-22-populating-your-dhcp-server-with-subnet-data-from-foreman.md:15]: typo: "to _do_ it"

**Comment by GregSutcliffe** [comment]
@timogoebel thanks! Interesting approach - I had assumed you'd used the `foreman()` function we ship in the puppet-foreman module. Other than having to parse the returned hash (awkward but possible) is there another reason bot to use `foreman()`? Since we ship it, it would be great to have some good real-world examples to point people at :p

**Comment by timogoebel** [comment]
> Since we ship it, it would be great to have some good real-world examples to point people at :p

I agree. I wasn't aware of `foreman()`. I'll update the blog post. The function should make things a lot easier. However my schedule for this week is incredibly full, to it might take some time.

**Comment by GregSutcliffe** [comment]
That's cool, it's worth the wait. I'll have a play around myself and see what I can come up with for DNS, since it's a similar problem space - it should be little more than a `.pp` file with a function call that we can use with puppet apply.

**Comment by dLobatog** [comment]
This has been open for a while, @GregSutcliffe @timogoebel how about just mentioning it's possible to use the `foreman()` function to do this and linking to the docs, while leaving the rest of the blogpost intact? This is worth publishing already IMOf

**Comment by GregSutcliffe** [comment]
+1 to a quick mention - @timogoebel?

**Comment by timogoebel** [comment]
I'd prefer to rewrite this to `foreman()`. Hope, I find time soon. Things are a little off the hook here...

**Comment by mmoll** [comment]
@timogoebel any chance to complete this in the near future?

**Comment by GregSutcliffe** [comment]
I'm inclined to just merge this with the suggested note from @dLobatog. I think it's still accurate / relevant, right @timogoebel / @dLobatog / @mmoll ?

File: _posts/2016-07-22-populating-your-dhcp-server-with-subnet-data-from-foreman.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+  # get subnet definitions via Foreman API and create dhcp:pool resources accordingly
+  service_foreman_create_dhcp_pools()
+```
+
+Finally we need to add the custom function, that does the heavy lifting, to our puppet module. Place it under `lib/puppet/parser/functions/service_foreman_create_dhcp_pools.rb`.
+The code currently filters the subnets based on the location of the subnet and your dhcp server.
+
+```ruby
+require 'apipie-bindings'
```
Comment (ekohl) [inline _posts/2016-07-22-populating-your-dhcp-server-with-subnet-data-from-foreman.md:88]: We no longer ship the `apipie-bindings` in the AIO install. Could this be rewritten to [foreman()](https://github.com/theforeman/puppet-foreman/blob/master/lib/puppet/functions/foreman/foreman.rb) and a Puppet 4 loop?

**Review by ekohl — COMMENTED** [review]
This is still a very valuable blog IMHO.

**Comment by tbrisker** [comment]
Ping @timogoebel @GregSutcliffe any chance of getting this merged any time? Maybe @ekohl can send a pr updating this to use `foreman()` etc?

**Comment by tbrisker** [comment]
@timogoebel looks like it's been 3 years since you last touched this, i'll assume you don't plan on finishing it and close it. If you or anyone else who's been involved wants to pick it up than by all means feel free to reopen.

**Comment by timogoebel** [comment]
@tbrisker: Thanks for closing this. Who am I kidding, I won't have time to revisit this anytime soon.

**Pull request closed without merging**